store/store.go

```diff
@@ -1018,7 +1018,8 @@ func (s *Store) Assertion(assertType *asserts.AssertionType, primaryKey []string
 	defer resp.Body.Close()
 
 	if resp.StatusCode != 200 {
-		if resp.Header.Get("Content-Type") == "application/json" {
+		contentType := resp.Header.Get("Content-Type")
+		if contentType == "application/json" || contentType == "application/problem+json" {
```

Comment:
@mvo5 because I saw I was getting the wrong kind of errors when the spread tests were failing, SAS is returning problem+json not just json